src/lib/Libifl/pbsD_locjob.c

```diff
@@ -134,3 +126,47 @@ __pbs_locjob(int c, char *jobid, char *extend)
 
 	return ploc;
 }
+
+/**
+ * @brief
+ *      This function does the LocateJob request.
+ *
+ * @param[in] c - connection handler
+ * @param[in] jobid - job identifier
+ * @param[in] extend - string to encode req
+ *
+ * @return	string
+ * @retval	destination name	success
+ * @retval	NULL	error
+ */
+char *
+__pbs_locjob(int c, char *jobid, char *extend)
+{
+	if ((jobid == NULL) || (*jobid == '\0')) {
+		pbs_errno = PBSE_IVALREQ;
+		return NULL;
+	}
+
+	if (multi_svr_op(c)) {
+		svr_conn_t **svr_conns = get_conn_svr_instances(c);
+		int start = 0;
+		int ct;
+		int nsvrs = get_num_servers();
+		char *ret = NULL;
+		int i;
+
+		if ((start = get_obj_location_hint(jobid, MGR_OBJ_JOB)) == -1)
+		    start = 0;
+
+		for (i = start, ct = 0; ct < nsvrs; i = (i + 1) % nsvrs, ct++) {
+			if (!svr_conns[i] || svr_conns[i]->state != SVR_CONN_STATE_UP)
+				continue;
+			ret = pbs_locjob2(svr_conns[i]->sd, jobid, extend);
+			if (ret)
```

Comment:
If you are getting this return value correctly, then we might be recording the internal server moves into the tracking table. Should we do that? That might be a huge no of jobs to track and what's the point if they are part of the same cluster?
Instead, we should record into the tracking table only if the job moves to a remote cluster.
The problem now would be that if a qmove happens after the internal move; pbs_locjob will fail to report that.

Reply:
I don't think we support qmove between servers of the same cluster (especially with PBS_SERVER values being the same, I don't see how it'll work). So, can we wait until we add support for qmove between internal servers (if at all)? If not, please let me know if you have a suggestion, I don't think we can handle it without knowing how qmove between internal servers will work.

Reply:
PBS_SERVER values can be distinct as well between instances in a multi-server right?
I was talking about the internal move that happens when the scheduler decides to run the job in a different instance.

Reply:
> PBS_SERVER values can be distinct as well between instances in a multi-server right?
Even if PBS_SERVER is different, if they are on the same host, they will all resolve to the same hostname, so it still cannot work.

> I was talking about the internal move that happens when the scheduler decides to run the job in a different instance.
If a job is moved from one multi-server to another, do we keep its information around in db? I thought we delete it from the original server instance. If we don't, I think we should just delete it as the job is still in the same queue, so moving from one multi-server to another isn't technically a qmove. Refer to the definition of qmove from the ref guide:

_"2.48 qmove
Moves a PBS job from one queue to another
"_

Reply:
> Even if PBS_SERVER is different, if they are on the same host, they will all resolve to the same hostname, so it still cannot work.

What if they are on different hosts?

>  so moving from one multi-server to another isn't technically a move

Agreed! That's what I'm trying to say. 
As you are enhancing pbs_locjob for multi-svr, we should make sure that internal moves are not being tracked. If we do so, pbs_locjob could still return the wrong internal server instead of a remote server for a multi-svr case.

Reply:
ok, I see your concern, and I'd have loved to test it out, but right now qmove between a cluster with multi-server and a cluster with single server is broken, so I cannot test out whether this works or not. But, the point that I'm trying to make is, if it doesn't work, then it'd be something to fix on the server side, not pbs_locjob. The server should delete a job after it's moved to another instance in the local cluster, and then the changes that this PR adds will make sure that all instances are sent the request to know which local/remote server the job was moved to. Does that make sense?

Reply:
Interesting conversation! When the qmove across complexes issue is resolved, we should test what happens when a job is moved from a single server cluster to a multi-server cluster and then moved locally by the scheduler. Does the pulling complex report back the finished status of the job? and does pbs_locjob issued on furnishing complex work in this scenario?

Reply:
ya, that'd be interesting to test .. i'm not completely sure how it'll work since the single server cluster won't know much about the multi-server cluster .. this is something that we'll probably not be able to support in the first version.